WcaOnRails/spec/factories/competitions.rb

```diff
@@ -44,5 +44,10 @@
       with_delegate
       isConfirmed true
     end
+
+    trait :confirmed_and_visible do
```

Comment:
I think/hope FactoryGirl manages things so each trait is only called once.

If not, this might not be worth changing.